Server/mods/deathmatch/logic/CPlayerManager.cpp

```diff
@@ -74,327 +67,139 @@ void CPlayerManager::PulseZombieCheck()
 {
-    // Delete all the items in the list
-    while (!m_Players.empty())
-        delete *m_Players.begin();
+    return m_JoinedByBitStreamVer.CountJoined(); // O(n), where n = m_JoinedByBitStreamVer.CountUniqueVersions()
 }
 
-void CPlayerManager::BroadcastOnlyJoined(const CPacket& Packet, CPlayer* pSkip)
+// Find player by fully matching nick. Todo: Partial nick
+CPlayer* CPlayerManager::Get(const char* szNick, bool bCaseSensitive) const
 {
-    // Make a list of players to send this packet to
-    CSendList sendList;
-
-    // Send the packet to each ingame player on the server except the skipped one
-    list<CPlayer*>::const_iterator iter = m_Players.begin();
-    for (; iter != m_Players.end(); iter++)
-    {
-        CPlayer* pPlayer = *iter;
-        if (pPlayer != pSkip && pPlayer->IsJoined())
-        {
-            sendList.push_back(pPlayer);
-        }
-    }
-
-    CPlayerManager::Broadcast(Packet, sendList);
-}
-
-void CPlayerManager::BroadcastDimensionOnlyJoined(const CPacket& Packet, ushort usDimension, CPlayer* pSkip)
-{
-    // Make a list of players to send this packet to
-    CSendList sendList;
-
-    // Send the packet to each ingame player on the server except the skipped one
-    list<CPlayer*>::const_iterator iter = m_Players.begin();
-    for (; iter != m_Players.end(); iter++)
-    {
-        CPlayer* pPlayer = *iter;
-        if (pPlayer != pSkip && pPlayer->IsJoined())
-        {
-            if (pPlayer->GetDimension() == usDimension)
-                sendList.push_back(pPlayer);
-        }
-    }
-
-    CPlayerManager::Broadcast(Packet, sendList);
-}
-
-void CPlayerManager::BroadcastOnlySubscribed(const CPacket& Packet, CElement* pElement, const char* szName, CPlayer* pSkip)
-{
-    // Make a list of players to send this packet to
-    CSendList sendList;
-
-    // Send the packet to each ingame player on the server except the skipped one
-    list<CPlayer*>::const_iterator iter = m_Players.begin();
-    for (; iter != m_Players.end(); iter++)
-    {
-        CPlayer* pPlayer = *iter;
-        if (pPlayer != pSkip && pPlayer->IsJoined() && pPlayer->IsSubscribed(pElement, szName))
-        {
-            sendList.push_back(pPlayer);
-        }
-    }
-
-    CPlayerManager::Broadcast(Packet, sendList);
-}
-
-// Send one packet to a list of players, grouped by bitstream version
-static void DoBroadcast(const CPacket& Packet, const std::multimap<ushort, CPlayer*>& groupMap)
-{
-    if (!CNetBufferWatchDog::CanSendPacket(Packet.GetPacketID()))
-        return;
-
-    // Use the flags to determine how to send it
-    NetServerPacketReliability Reliability;
-    unsigned long              ulFlags = Packet.GetFlags();
-    if (ulFlags & PACKET_RELIABLE)
-    {
-        if (ulFlags & PACKET_SEQUENCED)
-        {
-            Reliability = PACKET_RELIABILITY_RELIABLE_ORDERED;
-        }
+    const auto iter = std::find_if(m_Players.begin(), m_Players.end(), [bCaseSensitive, szNick](CPlayer* pPlayer) {
```

Comment:
If `m_Players` only contains joined players, doesn't this break `getPlayerFromName` (in onPlayerConnect)?

Reply:
Sorry. It turns out the comment for `m_Players` has wrong wording. It should say `all connected players`, not `all joined players`. My bad.